test/test_tensorclass_nofuture.py

```diff
@@ -533,20 +729,29 @@ class MyDataParent:
     X_clone = X.clone()
     td = TensorDict({}, batch_size)
     td_clone = td.clone()
-    data_nest = MyDataNest(X=X, batch_size=batch_size)
-    data = MyDataParent(X=X, y=data_nest, z=td, W=W, batch_size=batch_size)
+    data_nest = MyDataNest(X=X, v="test_nested", batch_size=batch_size)
```

Comment:
please move the nested-related stuff at the bottom (before checks on batch size)  to not split "X" test

Reply:
please also add comments like:
`# check set attr for tensor`
`# check set attr nested class`
etc...